genesyscloud/routing_settings/genesyscloud_routing_settings_proxy.go

```diff
@@ -0,0 +1,139 @@
+	return updateRoutingSettingsTranscriptionFn(ctx, p, transcriptionSettings)
+}
+
+func getRoutingSettingsFn(ctx context.Context, p *routingSettingsProxy) (*platformclientv2.Routingsettings, *platformclientv2.APIResponse, error) {
+	settings, resp, err := p.routingSettingsApi.GetRoutingSettings()
+	if err != nil {
+		return nil, resp, fmt.Errorf("error retrieving routing settings %s", err)
+	}
+	return settings, resp, nil
+}
+
+func updateRoutingSettingsFn(ctx context.Context, p *routingSettingsProxy, routingSettings *platformclientv2.Routingsettings) (*platformclientv2.Routingsettings, *platformclientv2.APIResponse, error) {
+	settings, resp, err := p.routingSettingsApi.PutRoutingSettings(*routingSettings)
+	if err != nil {
+		return nil, resp, fmt.Errorf("error updating routing settings %s", err)
+	}
+	return settings, resp, nil
+}
+
+func deleteRoutingSettingsFn(ctx context.Context, p *routingSettingsProxy) (*platformclientv2.APIResponse, error) {
+	resp, err := p.routingSettingsApi.DeleteRoutingSettings()
+	if err != nil {
+		return resp, fmt.Errorf("error deleting routing settings %s", err)
+	}
+	return resp, nil
+}
+
+func getRoutingSettingsContactCenterFn(ctx context.Context, p *routingSettingsProxy) (*platformclientv2.Contactcentersettings, *platformclientv2.APIResponse, error) {
+	contactCenter, resp, getErr := p.routingSettingsApi.GetRoutingSettingsContactcenter()
+	if getErr != nil {
+		return nil, resp, fmt.Errorf("failed to read Contact center for routing setting %s", getErr)
+	}
+	return contactCenter, resp, nil
+}
+
+func updateRoutingSettingsContactCenterFn(ctx context.Context, p *routingSettingsProxy, contactCenterSettings platformclientv2.Contactcentersettings) (*platformclientv2.APIResponse, error) {
+	resp, err := p.routingSettingsApi.PatchRoutingSettingsContactcenter(contactCenterSettings)
+	if err != nil {
+		return resp, fmt.Errorf("failed to update transcription for routing setting %s", err)
+	}
+	return resp, nil
+}
+
+func getRoutingSettingsTranscriptionFn(ctx context.Context, p *routingSettingsProxy) (*platformclientv2.Transcriptionsettings, *platformclientv2.APIResponse, error) {
+	transcription, resp, getErr := p.routingSettingsApi.GetRoutingSettingsTranscription()
+	if getErr != nil {
+		return nil, resp, fmt.Errorf("failed to read transcription for routing setting %s", getErr)
+	}
+	return transcription, resp, nil
+}
+
+func updateRoutingSettingsTranscriptionFn(ctx context.Context, p *routingSettingsProxy, transcriptionSettings platformclientv2.Transcriptionsettings) (*platformclientv2.Transcriptionsettings, *platformclientv2.APIResponse, error) {
+	transcription, resp, err := p.routingSettingsApi.PutRoutingSettingsTranscription(transcriptionSettings)
+	if err != nil {
+		return nil, resp, fmt.Errorf("failed to update transcription for routing setting %s", err)
+	}
+	return transcription, resp, nil
+}
```

Comment:
For all these implementation functions, I would just return the API function call directly since you're already checking the error and adding error context in `resource_genesyscloud_routing_settings.go`.